pyuvm/s21_pyuvm_reg_map.py

```diff
@@ -46,117 +47,116 @@ def get_registers(self):
-            local_bus_op.addr       = reg_address 
-            local_bus_op.n_bits     = self._regs[reg_address].get_n_bits()
-            local_bus_op.byte_en    = local_adapter.get_byte_en()
-            ## Parse the local bus operation with the adapter
+                uvm_not_implemeneted(self.header, "EXPLICIT_PREDICTION not implemented")
+
+    # process_read_operation
+    async def process_read_operation(self, reg_address, path: path_t, check: check_t):
+        # Get the sequencer and the adapter
+        local_adapter = self.get_adapter()
+        local_sequencer = self.get_sequencer()
+        # check if the Path is set to BACKDOOR, FRONTDOOR or USER_FRONTDOOR
+        if (path is path_t.BACKDOOR):
+            uvm_not_implemeneted(self.header, "BACKDOOR not implemented")
+        elif (path is path_t.USER_FRONTDOOR):
+            uvm_not_implemeneted(self.header, "USER_FRONTDOOR not implemented")
+        elif (path is path_t.FRONTDOOR):
+            # Populate internal Item
+            local_bus_op = uvm_reg_bus_op()
+            local_bus_op.kind = access_e.uvm_READ
+            local_bus_op.addr = reg_address
+            local_bus_op.n_bits = self._regs[reg_address].get_n_bits()
+            local_bus_op.byte_en = local_adapter.get_byte_en()
+            # Parse the local bus operation with the adapter
             local_adapter.reg2bus(local_bus_op)
-            ## The get parent sequence is still not implemented hence we should be using the base sequence type for the time being 
-            local_sequence          = local_adapter.get_parent_sequence()
-            ## Start the sequence on local sequencer
+            # The get parent sequence is still not implemented hence we
+            # hould be using the base sequence type for the time being
+            local_sequence = local_adapter.get_parent_sequence()
+            # Start the sequence on local sequencer
             await local_sequence.start(local_sequencer)
-            ## Get the sequence item from the local sequence
-            local_item              = uvm_sequence_item
+            # Get the sequence item from the local sequence
+            local_item = uvm_sequence_item
             local_sequence.copy(local_item)
-            ## Assign the response and read data back
+            # Assign the response and read data back
             local_adapter.bus2reg(local_item, local_bus_op)
-            ## Invoke the prediction
-            if enable_auto_predict == True:
-                local_predictor     = self.get_predictor() 
+            # Invoke the prediction
+            if (enable_auto_predict is True):
+                local_predictor = self.get_predictor()
                 local_predictor.predict(local_bus_op, check)
             else:
-                error_out(self.header,"EXPLICIT_PREDICTION not implemented")
-
-
+                uvm_not_implemeneted(self.header, "EXPLICIT_PREDICTION not implemented")
```

Comment:
up to here

- replaced `error_out` with `uvm_error` or `uvm_not_implemented` accordingly 
- replaced the `_pyuvm_` with `_uvm_` 
- fixed flake8 errors 
- all the `PYUVM_` enum values have been replaced with `UVM_`